constraints/src/main/java/gov/nasa/jpl/aerie/constraints/tree/WindowsValue.java

```diff
@@ -0,0 +1,51 @@
+package gov.nasa.jpl.aerie.constraints.tree;
+
+import gov.nasa.jpl.aerie.constraints.model.EvaluationEnvironment;
+import gov.nasa.jpl.aerie.constraints.model.LinearEquation;
+import gov.nasa.jpl.aerie.constraints.model.LinearProfile;
+import gov.nasa.jpl.aerie.constraints.model.SimulationResults;
+import gov.nasa.jpl.aerie.constraints.time.Interval;
+import gov.nasa.jpl.aerie.constraints.time.Segment;
+import gov.nasa.jpl.aerie.constraints.time.Windows;
+import gov.nasa.jpl.aerie.merlin.protocol.types.Duration;
+
+import java.util.Objects;
+import java.util.Set;
+
+public final class WindowsValue implements Expression<Windows> {
+  public final boolean value;
+
+  public WindowsValue(final boolean value) {
+    this.value = value;
+  }
+
+  @Override
+  public Windows evaluate(final SimulationResults results, final Interval bounds, final EvaluationEnvironment environment) {
+    return new Windows(bounds, value);
+  }
+
+  @Override
+  public void extractResources(final Set<String> names) { }
+
+  @Override
+  public String prettyPrint(final String prefix) {
+    return String.format(
+        "\n%s(value %s)",
+        prefix,
+        this.value
+    );
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (!(obj instanceof WindowsValue)) return false;
+    final var o = (WindowsValue)obj;
+
+    return Objects.equals(this.value, o.value);
+  }
+
+  @Override
+  public int hashCode() {
```

Comment:
I'd also suggest making this a `record` class to avoid the equals/hashcode boilerplate.